### ISSUE TITLE
Document IsImplicitlyDefined

### ISSUE THREAD

**Issue by zivkan** [NuGet/Home#13529]
Document the `IsImplicitlyDefined` metadata on `PackageReference` items, that MSBuild SDKs are expected to set to any PackageReference the SDK itself defines, so that Visual Studio's PackageManager doesn't try to offer it as an upgradable package that will fail if the upgrade is attempted.  It also makes the PackageReference work for projects using CPM.

**Comment by nkolev92** [comment]
Another notable thing here is that we should define the behavior of CPM and IsImplicitlyDefined 

Related: https://github.com/NuGet/Home/issues/11967#issuecomment-2039734833

**Comment by Evangelink** [comment]
It would be good to also clear out whether or not we need the `VersionOverride` setting when using `IsImplictlyDefined` (doesn't seem to be the case)

**Comment by Evangelink** [comment]
I don't think this should be closed, the documentation clearly need to be improved here.
